libr/bin/p/bin_bflt.c

```diff
@@ -0,0 +1,303 @@
+		n_got = get_ngot_entries (obj);
+		n_reloc += n_got;
+	}
+	bflt_reloc_table = calloc (1, n_reloc * sizeof (struct bflt_relocation_t));
+	if (!bflt_reloc_table) {
+		return list;
+	}
+	if (obj->hdr->flags & FLAT_FLAG_GOTPIC) {
+		ut32 offset;
+		for (offset = 0, i = 0; i < n_got; i++, offset += 4) {
+			ut32 got_entry;
+			len = r_buf_read_at (obj->b, obj->hdr->data_start + offset, (ut8 *)&got_entry, sizeof (ut32));
+			got_entry = r_swap_ut32 (got_entry);
+			if (VALID_GOT_ENTRY (got_entry)) {
+				bflt_reloc_table[i].addr_to_patch = got_entry;
+				bflt_reloc_table[i].data = got_entry + BFLT_HDR_SIZE;
+			} else {
+				break;
+			}
+		}
+	}
+
+	for(i = 0; i < obj->hdr->reloc_count; i++) {
+		ut32 reloc_pointer;
+
+		len = r_buf_read_at (obj->b, obj->hdr->reloc_start + (i * 4), (ut8 *)&reloc_pointer, 4);
+		if (len != 4) {
+			return list;
+		}
+
+		reloc_pointer = r_swap_ut32 (reloc_pointer);
+		ut32 reloc_offset = reloc_pointer + BFLT_HDR_SIZE;
+
+		if (reloc_offset < obj->hdr->bss_end) {
+			ut32 reloc_fixed;
+			ut32 reloc_data_offset;
+
+			len = r_buf_read_at (obj->b, reloc_offset, (ut8 *)&reloc_fixed, 4);
+			reloc_data_offset = reloc_fixed + BFLT_HDR_SIZE;
+
+			bflt_reloc_table[n_got + i].addr_to_patch = reloc_offset;
+			bflt_reloc_table[n_got + i].data = reloc_data_offset;
+
+			RBinReloc *reloc = R_NEW0 (RBinReloc);
+			reloc->type = R_BIN_RELOC_32;
+			reloc->paddr = bflt_reloc_table[n_got + i].addr_to_patch;
+			reloc->vaddr = reloc->paddr;
+			r_list_append (list, reloc);			
+		}
+
+	}
+	obj->bflt_reloc_table = bflt_reloc_table;
+	return list;
+}
+
+static RBinInfo *info(RBinFile *arch) {
+	struct r_bin_bflt_obj *obj = (struct r_bin_bflt_obj*)arch->o->bin_obj;
+	RBinInfo *info = R_NEW0(RBinInfo);
+
+	if (!info) {
+		return NULL;
+	}
+	info->file = arch->file ? strdup (arch->file) : NULL;
+	info->rclass = strdup ("bflt");
+	info->bclass = strdup ("bflt" );
+	info->type = strdup ("bFLT (Executable file)");
+	info->os = strdup ("Linux");
+	info->subsystem = strdup ("Linux");
+	info->arch = strdup ("arm");
+	info->big_endian = obj->endian;
+	info->bits = 32;
+	info->has_va = false;
+	info->dbg_info = 0;
+	info->machine = strdup ("unknown");
+
+	return info;
+}
+
+static int check_bytes(const ut8 *buf, ut64 length) {
+	return true;
+}
+
+static int check(RBinFile *arch) {
+	/*const ut8 *bytes = arch ? r_buf_buffer (arch->buf) : NULL;
+	ut64 sz = arch ? r_buf_size (arch->buf): 0; */
+	return true;
+}
+
+static Sdb* get_sdb(RBinObject *o) {
+	if (!o) {
+		return NULL;
+	}
+	struct r_bin_bflt_obj *bin = (struct r_bin_bflt_obj *) o->bin_obj;
+	if (bin->kv) {
+		return bin->kv;
+	}
+
+	return NULL;
+}
+
+static int destroy(RBinFile *arch) {
+	r_bin_bflt_free ((struct r_bin_bflt_obj*)arch->o->bin_obj);
+	return true;
+}
+
+static ut64 baddr(RBinFile *arch) {
+	return 0;
+}
+
+static RList *sections(RBinFile *arch) {
+	return NULL;
+}
+
+static RList *symbols(RBinFile *arch) {
+	return NULL;
+}
+
+static RList *fields(RBinFile *arch) {
```

Comment:
you can set the function pointer to NULL to avoid defining all those dummy functions that return nothing